RevisionNotes/MIT_Lec12.md

```diff
@@ -0,0 +1,222 @@
+## Graphs, Networks, Incidence Matrices
+
+This session explores the linear algebra of electrical networks and sheds light on important results in graph theory.
+
+Consider this directed graph - 
+
+![Graph](Images/Graph_lec12.PNG)
+
+Here `m` = 5 edges, `n` = 4 nodes.
+
+### Incidence Matrix
+
+`A` - Incidence matrix used to denote this graph.
+
+It is the 5 by 4 matrix which tells us which nodes are connected by which edges.
+
+In general, an incidence matrix is a logical matrix that shows the relationship between two classes of objects, usually called an incidence relation. Here, we are showing the relationship between the edges and nodes of the graph.
+
+![Incidence Matrix](Images/Matrix_A_lec12.PNG)
+
+Row numbers in A are edge numbers, column numbers are node numbers.
+
+* -1 means the edge is going out of the node.
+
+* 1 means the edge is going into the node.
+
+* 0 means the edge does not connect this node.
+
+### Analysing the Incidence Matrix
+
+#### <u>Elimination</u>
+
+After elimination, we get -
+
+![After Elimination](Images/Elimination_lec12.PNG)
+
+This represents the following <b>tree</b>.
+
+![Tree](Images/GraphToTree_lec12.PNG)
+
+Elimination reduces every graph to a tree (the graph has no closed loops). Rows are dependent when edges form a loop. Independent rows come from trees.
+
+This also gives us the rank, `r` = 3.
+
+#### <u>The Nullspace</u>
+
+Let the nullspace solution of `A` be `x`.
+
+`x` is basically the potential at each node.
+
+`Ax = 0`
+
+![Nullspace of A](Images/Nullspace_lec12.PNG)
+
+![Potential Difference](Images/PotentialDiff_lec12.PNG)
+
+`Ax` denotes the Potential Difference between nodes.
+
+It is 0 when all the potentials are the same.
+
+So nullspace includes all vectors of the form
+`c(1, 1, 1, 1)`
+
+So the <b>dimension of nullspace = `n - r` = 1</b>
+
+Current doesn't flow when potential difference is 0.
+
+We can raise or lower all voltages by the same amount `c`, without changing the differences. There is an "arbitrary constant" in the voltages. 
+
+The nullspace disappears when we fix x<sub>4</sub> = 0. The unknown x<sub>4</sub> is removed and so is the fourth column of A (the column multiplied x<sub>4</sub>). Basically, node 4 has been "grounded." So only zero vector is in the nullspace now.
+
+#### <u>The Row Space</u>
+
+The row space contains all combinations of the five rows.
```

Comment:
Can you please specify that "row space contains all combinations of 3 row basis vectors since the matrix has a rank of 3". "all combinations of five rows" seems a bit dicey to me since all 5 rows are technically not row basis vectors.

Reply:
Yes, corrected.